packages/dds/matrix/src/matrix.ts

```diff
@@ -276,7 +337,14 @@ export class SharedMatrix<T = any>
 		this.cells.setCell(rowHandle, colHandle, value);
 
 		if (this.isAttached()) {
-			this.sendSetCellOp(row, col, value, rowHandle, colHandle);
+			this.protectAgainstReentrancy(() =>
+				this.sendSetCellOp(row, col, value, rowHandle, colHandle),
+			);
+		}
+
+		// Avoid reentrancy by raising change notifications after the op is queued.
+		for (const consumer of this.consumers.values()) {
+			consumer.cellsChanged(row, col, 1, 1, this);
```

Comment:
protectAgainstReentrancy() protection should be all around this code - I'd wrap almost whole function